### ISSUE TITLE
Can we have `start` or `begin` to go with `end`?

### ISSUE THREAD

**Issue by trefethen** [chebfun/chebfun#1554]
In Chebfun you can write e.g.
```
>> f = chebfun('exp(x)');
>> f(end)
ans =  2.718281828459046
```
It seems to me there should be an analogous way to get the initial value of `f`, such as `f(start)` or `f(begin)`.  Indeed I recalled discussing this some years ago, but I don't recall what happened.  Can somebody remind me if we've already got this, or implement it if we don't?

Thanks.

**Comment by mhsnjvd** [comment]
Matlab doesn't have such syntax either: it offers 'end' but no 'start' or 'begin'. One could use
```
f(f.domain)
f(f.domain(1))
```
but this is not as simple as f(end).

**Comment by trefethen** [comment]
Yes, and of course the reason Matlab doesn't have such syntax is that every vector begins at 1.

**Comment by jrus** [comment]
This is one of the places where semantics for a discrete array starting at 1 don’t cleanly map to the case of an arbitrary continuous interval. In Matlab, the goal of `end` is to allow counting back from the end, e.g. to pluck out the last few elements of an array (`a(end-4:end)`), or to slice off the first and last elements (`a(2:end-1)`), etc.

Chebfun doesn’t really have a syntactic analog, because there’s no similar shorthand for representing a continuous interval as a matlab object. To do the same kind of operation takes constructing a new Chebfun with a different domain. (At least, as far as I can tell.)

**Comment by davidagross** [comment]
If the issue of whether to implement this resolves positively, might "start", "begin", or "cheb.start", or "cheb.begin" simply be a function handle that evaluates to f.domain(1) before being used in a deeply internal feval?

Also @jrus, I think you can evaluate f at a chebfun of new values to change the domain, no?

**Comment by ajt60gaibb** [comment]
The analogue to `A(j:k)` in Chebfun is `f{a,b}`, which restricts `f` to the interval `[a,b]`.

**Comment by jrus** [comment]
Aha, thanks! I didn’t know about that one. Does the `end` keyword work in that context, and can it be used in an expression like `f{end-2, end-1}`? I assume a proposed `begin` or `start` would work there as well?

The curly brace interval notation does seem to get one brief and not-all-that-explicit mention in the Chebfun guide in chapter 4, from a line in the middle of other exposition. Among the first 6 (searchable pdf) chapters of the ATAP book, one exercise uses this notation without explanation, but there’s no other mention of it there.

Aside: It’s a bit off topic for this thread, but one thing that would really help me personally with use of Chebfun is more comprehensive API documentation. I have a large pile of tiny questions about possible options to use with different constructors, behavior in corner cases, etc., and the main place to get them answered seems to be the code itself. I feel awkward about asking each one individually, because it seems like it would overwhelm whoever is trying to answer. The Matlab interactive help is somewhat useful, but is not comprehensive and is a bit hard to read all at once. There are nice comments in some parts of the code, and the code itself is mostly fairly clear, but there are hundreds (thousands?) of matlab files here, so it’s a bit hard to figure out exactly where particular behavior would be defined and where to look for it.
